api/src/main/java/org/openmrs/api/impl/PatientServiceImpl.java

```diff
@@ -264,6 +264,20 @@ public List<Patient> getPatients(String name, String identifier, List<PatientIde
 		return Context.getPatientService().getPatients(name, identifier, identifierTypes, matchIdentifierExactly, 0, null);
 	}
 	
+	/**
+	 * @see PatientService#getPatientsByIdentifier(org.openmrs.api.PatientService)
+	 */
+	@Override
+	@Transactional(readOnly = true)
+	public List<Patient> getPatientsByIdentifier(String name, String identifier, List<PatientIdentifierType> identifierTypes,
+	        boolean matchIdentifierExactly) throws APIException {
+		if (identifierTypes == null) {
```

Comment:
you want to avoid checking null instead do something like

```
if(CollectionUtils.isEmpty(identifierTypes )){
  identifierTypes = Collections.emptyList();
 }
```